### PR TITLE
Fix: expanded form view row insert bug

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

- Adding a new row should reflect in the grid view
- After adding new row LTAR data should be present in expanded form
- Unlinking LTAR should remove it from the expanded form
- Removing the LTAR link from any page navigating back to page `1`
- Add a new row from the list should link to the parent ( new record form )
- Unsaved row saving on switching table

re #3565 #3566

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/nc-gui/components/smartsheet/Grid.vue [hunk 1]
```diff
@@ -324,6 +324,35 @@ const showContextMenu = (e: MouseEvent, target?: { row: number; col: number }) =
     contextMenuTarget.value = target
   }
 }
+
+const rowRefs = $ref<any[]>()
+
+/** save/update records before unmounting the component */
+onBeforeUnmount(async () => {
+  console.log(rowRefs)
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/smartsheet/Grid.vue:332]: Log can be removed

File: packages/nc-gui/components/smartsheet/Grid.vue [hunk 2]
```diff
@@ -490,9 +519,9 @@ const showContextMenu = (e: MouseEvent, target?: { row: number; col: number }) =
             </SmartsheetRow>
 
             <!--
-      TODO: add relationType !== 'bt' ?
-      v1: <tr v-if="!isView && !isLocked && !isPublicView && isEditable && relationType !== 'bt'">
-    -->
+TODO: add relationType !== 'bt' ?
+v1: <tr v-if="!isView && !isLocked && !isPublicView && isEditable && relationType !== 'bt'">
+-->
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/smartsheet/Grid.vue:524]: Formatting seems broken

File: packages/nc-gui/components/tabs/Smartsheet.vue [hunk 3]
```diff
@@ -29,11 +33,12 @@ const el = ref<typeof SmartsheetGrid>()
...
 
-const tabMeta = inject(
-  TabMetaInj,
-  computed(() => ({} as TabItem)),
-)
-const meta = computed<TableType>(() => metas.value?.[tabMeta?.value?.id as string])
+// const tabMeta = inject(
+//   TabMetaInj,
+//   computed(() => ({} as TabItem)),
+// )
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/tabs/Smartsheet.vue:39]: Can be removed I assume?

File: packages/nc-gui/components/virtual-cell/BelongsTo.vue [hunk 4]
```diff
@@ -24,7 +24,7 @@ const ListItems = defineAsyncComponent(() => import('./components/ListItems.vue'
 
 const column = inject(ColumnInj)!
 
-const reloadTrigger = inject(ReloadViewDataHookInj)!
+const reloadRowTrigger = inject(ReloadRowDataHookInj, createEventHook())!
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/virtual-cell/BelongsTo.vue:27]: `!` can be removed if you provide a default value (i.e. `createEventHook()`)

File: packages/nc-gui/components/virtual-cell/components/ListItems.vue [hunk 5]
```diff
@@ -109,9 +116,9 @@ const newRowState = computed(() => {
         <div class="flex-1" />
         <MdiReload class="cursor-pointer text-gray-500 nc-reload" @click="loadChildrenExcludedList" />
         <!--        Add new record -->
-        <a-button v-if="!isPublic" type="primary" size="small" @click="expandedFormDlg = true">{{
-          $t('activity.addNewRecord')
-        }}</a-button>
+        <a-button v-if="!isPublic" type="primary" size="small" @click="expandedFormDlg = true"
+          >{{ $t('activity.addNewRecord') }}
+        </a-button>
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/virtual-cell/components/ListItems.vue:121]: Formatting is off

**Review by bcakmakoglu — CHANGES_REQUESTED** [review]

**Review by bcakmakoglu — APPROVED** [review]

**Review by wingkwong — APPROVED** [review]

**Pull request merged**